a_Calculate_Centers/src/calculate_centers_HUC4.R

```diff
@@ -0,0 +1,146 @@
+#' @title Calcuate POI center for NHDPlusHR lakes by HUC4
+#' 
+#' @description
+#' for each HUC4, download the HR NHDPlus waterbody file, subset to lakes/res/
+#' impoundments, subset to >= 1ha, and calculate POI for each polygon
+#' 
+#' @param HUC4 text string; 4-digit huc from NHDPlus
+#' 
+#' @returns silently saves .csv file in mid folder of the POI centers and 
+#' associated WBD metadata
+#' 
+#' 
+calculate_centers_HUC4 <- function(HUC4) {
+  #set timeout for longer per issue #341: https://github.com/DOI-USGS/nhdplusTools/issues
+  options(timeout = 60000)
+  
+  # download the nhdplushr huc4, and note the filepath
+  fp <- download_nhdplushr(nhd_dir = "a_Calculate_Centers/nhd/", HUC4)
+  
+  # check to see if there are contents in the NHD Plus HR. If there aren't add
+  # that HUC to the list of empty hucs
+  if (length(list.files(fp)) == 0) {
+    empty <- read_lines("a_Calculate_Centers/out/empty_hucs.txt")
+    empty <- paste(HUC4, empty, sep = ", ")
+    write_lines(empty,
+                file.path("a_Calculate_Centers/out/",
+                          "empty_hucs.txt"))
+  } else { # otherwise, go through the process of calculaitng centers
+  
+    # open the NHDWaterbody layer, coerce to a {sf} object
+    wbd <- get_nhdplushr(fp, layers = "NHDWaterbody") %>%
+      bind_rows() %>% 
+      st_as_sf() 
+    
+    # filter the waterbodies for ftypes of interest. 390 = lake/pond; 436 = res;
+    # 361 = playa
+    wbd <- wbd %>% 
+      filter(FTYPE %in% c(390, 436, 361)) 
+    
+    # filter for area > 1 hectare (0.01 km^2)
+    wbd <- wbd %>% 
+      filter(AreaSqKM >= 0.01) 
+    
```

Comment:
suggestion: You could condense these lines here, not a huge deal but more concise

```suggestion
    wbd <- wbd %>% 
         filter(
        # filter the waterbodies for ftypes of interest. 390 = lake/pond; 436 = res;
        # 361 = playa 
             FTYPE %in% c(390, 436, 361),
       # ...and for area > 1 hectare (0.01 km^2)
             AreaSqKM >= 0.01) 
```